### ISSUE TITLE
Question: How to customize the mobile version of my site?

### ISSUE THREAD

**Issue by Tyrael84** [silexlabs/Silex#453]
First, just want to thank all the developers and the community. This is an incredible editor. I really enjoy that it's downloadable, HTML, and that I've gotten answers about my questions within a matter of hours. Truly terrific!

My problem; I'm nearing the end of this website I've been working on and it looks fantastic. It loads incredibly fast compared to the Wix counterpart. All around it's great. 

At first I figured it wasn't possible to have a mobile version yet (which I wasn't thrilled about but my desktop site didn't look 'too' bad on mobile devices). Then I saw the mobile editor button and unfortunately I must have designed my site poorly because when I open it in mobile editor everything jumps all around the page. The header is at the bottom, the copyright at the top, and all the content is mismatched. Now this wouldn't be a problem if I could easily edit the mobile version like the desktop version but it seems this is not possible. When I drag something or change it's x/y it doesn't seem to want to move.

I can only assume that I didn't build the site properly (lack of < div . > tags?). If I have to I am willing to start over and build the mobile version first. Before I do that, however, I want to know what I did wrong so I can correct it or if I can correct this problem without starting from scratch.

TL;DR:

Mobile editor scrambles the entire site and it isn't possible to rearrange everything the way you can in the desktop version. What do?

Thanks,

Nick

:)

**Comment by lexoyo** [comment]
Hi Nick! Thanks for the kind words !! :D I'm a volunteer and the other contributors too :yellow_heart: 

Your question is good, let's start documenting this brand new functionality which took me 1 year to release :)

The mobile editor makes everything different, no more drag/drop or x/y, you are supposed to customize the mobile version with this:

* up/down arrows in the [context menu](https://github.com/silexlabs/Silex/wiki/context-menu) to move up and down the elements inside their container ![screenshot from 2016-08-03 23-21-30](https://cloud.githubusercontent.com/assets/715377/17382441/20c6e7b4-59d1-11e6-8505-feb75b5f2c17.png)

* show/hide elements from the mobile version with the "visible in mobile version" check box: 
![screenshot from 2016-08-03 23-25-24](https://cloud.githubusercontent.com/assets/715377/17382532/a661c330-59d1-11e6-9532-8a725a4fdb76.png)

* resize elements, change their styles (background color, borders...) like in the normal / desktop mode, they will center vertically and horizontally, they can arrange several elements on one line. Changing style in the mobile editor will not change the element's style in the desktop version, but do not change the structure of the site (text content, image, move into a container...)

* you can select  the hamburger like a normal element, and double click on it to change its image, or make it invisible in the mobile version with the visibility setting... 
![screenshot from 2016-08-03 23-29-57](https://cloud.githubusercontent.com/assets/715377/17382660/39419c2a-59d2-11e6-8e75-68e1651d33a5.png)

* style the mobile menu with CSS, it has the CSS class `.silex-pages` and contains the hamburger menu (`.menu-button`) and the list of all pages (a list of `.page-element` and `.page-link-active` when the page is opened, and the id `page-` + the name of the page) 

I tested this many times, and it seems to be enough to make a mobile version of any Silex website made for desktop... What do you think? Any question welcome!

**Comment by Tyrael84** [comment]
A-ha! The forward/backward buttons move the contents. Gotcha.

I suppose now my questions are:
Is it possible to upload a separate background image for the mobile version?
Is it possible to overlap images/items on top of each other?

If not I'm not sure how I'm going to make my current site mobile compatible unless I restart from the round up with the mobile version in mind. :\

**Comment by lexoyo** [comment]
> Is it possible to upload a separate background image for the mobile version?

yes, every style you change in the mobile version will not be seen in desktop version

> Is it possible to overlap images/items on top of each other?

no, but if you have something in a container, you can have a background image on the container

**Comment by Tyrael84** [comment]
Thanks for your help Alex you've been truly terrific. If I can give you
karma or something link me and I gladly will.

Sometimes when I edit things in the mobile editor (moving things up and
down) and switch back to the desktop some things (in the desktop) are
misarranged..

IE: Move container up in mobile viewer will cause it to be 'brought to
front' in the desktop version?

 Is this a bug?

**Comment by lexoyo** [comment]
you are very welcome :+1: 
your feedback is much appreciated

> Sometimes when I edit things in the mobile editor (moving things up and down) and switch back to the desktop some things (in the desktop) are misarranged.. IE: Move container up in mobile viewer will cause it to be 'brought to front' in the desktop version?

yes this is how it works: in mobile mode the order is the order of the html nodes, and in desktop the z-index is also the order of the html nodes... 

in desktop view you can also override the z-index with CSS so that the order is not used for z-index: in the element CSS you can add `z-index: 456` to explicitely set an order

![screenshot from 2016-08-08 11-06-59](https://cloud.githubusercontent.com/assets/715377/17474828/49d9db36-5d58-11e6-8976-cb8ca365741c.png)
![screenshot from 2016-08-08 11-06-45](https://cloud.githubusercontent.com/assets/715377/17474829/49f11c2e-5d58-11e6-8c0a-ae4e46feb0e8.png)

**Comment by Tyrael84** [comment]
Sorry for all the questions Alex...all these questions have been because of
this issue in the mobile editor mainly.

how do I remove the borders of this content box and the image at the
bottom. It's suppose to make it look like a ripped piece of paper but in
the mobile view I can't seem to position it closer to make it seamless. If
there's no way I can just hide it on mobile editor.

**Comment by lexoyo** [comment]
Hi
Mobile version is supposed to be simpler than desktop one so yes you can hide it
But please provide a link if you want me to tell you for sure

**Comment by Tyrael84** [comment]
Alex

I've found a bodge/work around to get the mobile editor to kind of work in
a much more convenient way. I've been using 'margin-top: -100px' (changing
-100px to whatever is necessary to get everything in the right position) in
the CSS and this has allowed me to 'sort of' move things around to the
point where I'm happy with my index page's desktop AND mobile version...BUT
in the CSS 'margin-top: ---px;' never 'saves' after you're done entering
the code so you have no idea what the margins are for each container and a
new problem now is when I make multiple pages (about, faq, etc) lining up
just the logo so it's in the same position as on the index page is nye
impossible...

If you have any ideas I'm really loosing hope. I've put way too many hours
into this new site and everything was going great but the mobile editor
seems to cause more problems then answers. I've re-created the site and
even while I constantly bob back and forth from mobile/desktop editor it's
almost impossible to get everything where you want it on all the pages. :\

**Comment by lexoyo** [comment]
Hello

I'm not sure I understand what you are trying to do nor what you are really doing :-/
When you add CSS, the best way to do it is [put a css class on your element and put the css in the css editor as described here](https://github.com/silexlabs/Silex/issues/184#user-content-properties-css-panel)

If you have an example online it may help me to understand what you are doing. And if you have suggestions for the mobile editor, try to be very specific, or do that after the storm has passed :)

**Comment by Tyrael84** [comment]
Attached an image to better explain


The gaps/margins/borders in the mobile editor are WAY to far apart to put
on a professional business site. Having to scroll an entire phone length to
get from the title to the content is terrible web design. I want 'Service
Areas' (and on other pages 'About', 'FAQ', 'etc) to be closer to the top of
the white and for the content below it to be closer to the title. The only
way I found I can move these in the mobile editor is to go into the CSS and
type "margin-top: -300px". 300 can obviously be replaced where needed. The
problem with this is the CSS does not 'save' the 'margin-top' field. It
applies it, but then it disappears from the CSS entry box.

What can I do from here :(

**Comment by lexoyo** [comment]
Did you try
- make the container smaller (it reduces the space between elements inside it)
- to add a css class to the element and put the style in the css
editor (this definitely saves the style no matter what)

**Comment by lexoyo** [comment]
hi
I just pushed a new feature to make it possible to hide elements from the desktop version
see on https://preprod.silex.me
